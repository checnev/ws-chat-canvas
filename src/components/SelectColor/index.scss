@import '~material-icons/iconfont/material-icons.scss';

.colorSelector {
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  margin: 0;

  &__colorSquare {
    @include material-icons();
    position: relative;
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;;
  }

  &__input {
    display: none;
  }

  &__input:checked + &__colorSquare {
    @include material-icon('check');
    font-size: 3em;
  }
}
