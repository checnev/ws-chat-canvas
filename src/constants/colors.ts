export type Color = string;
export const colors: Color[] = [
  '#ffd04e',
  '#ec7380',
  '#77dd77',
  '#50bade',
  '#ffe5b4',
  '#d191f9',
  '#3a9ff1',
  '#ffc2ce',
];
