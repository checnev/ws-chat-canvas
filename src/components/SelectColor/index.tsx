import React from 'react';
import cn from 'utils/cn';
import clsx from 'clsx';
import { colors } from 'constants/colors';
import './index.scss';

interface ColorSelectorProps extends React.InputHTMLAttributes<HTMLInputElement> {
  width?: number | string;
  height?: number | string;
}

class ColorSelector extends React.PureComponent<ColorSelectorProps> {
  static defaultProps: ColorSelectorProps = {
    name: 'color',
    width: '25%',
    height: 50,
  }
  render() {
    const colorSelector = cn('colorSelector');
    const { className, value, width, height, ...props } = this.props; 

    return (
      <fieldset className={clsx(className, colorSelector())}>
        {colors.map((color) => (
          <label className={colorSelector('label')} style={{ width }} key={color}>
            <input
              type="radio"
              className={colorSelector('input')}
              value={color}
              checked={color === value}
              {...props}
            />
            <span 
              className={colorSelector('colorSquare')}
              style={{
                backgroundColor: color,
                height,
              }}
            />
          </label>
        ))}
      </fieldset>
    )
  }
}

export default ColorSelector;
