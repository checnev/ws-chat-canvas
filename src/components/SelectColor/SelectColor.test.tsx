import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { colors } from 'constants/colors';
import SelectColor from 'components/SelectColor';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render SelectColor', () => {
  act(() => {
    render(<SelectColor onChange={() => {}} value={colors[0]} />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
