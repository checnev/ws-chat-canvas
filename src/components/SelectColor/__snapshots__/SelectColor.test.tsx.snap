// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SelectColor 1`] = `"<fieldset class=\\"colorSelector\\"><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#ffd04e\\" checked=\\"\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(255, 208, 78); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#ec7380\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(236, 115, 128); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#77dd77\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(119, 221, 119); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#50bade\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(80, 186, 222); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#ffe5b4\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(255, 229, 180); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#d191f9\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(209, 145, 249); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#3a9ff1\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(58, 159, 241); height: 50px;\\"></span></label><label class=\\"colorSelector__label\\" style=\\"width: 25%;\\"><input type=\\"radio\\" class=\\"colorSelector__input\\" name=\\"color\\" value=\\"#ffc2ce\\"><span class=\\"colorSelector__colorSquare\\" style=\\"background-color: rgb(255, 194, 206); height: 50px;\\"></span></label></fieldset>"`;
